File: Java/1428. Leftmost Column with at Least a One.java
/**
 * // This is the BinaryMatrix's API interface.
 * // You should not implement it, or speculate about its implementation
 * interface BinaryMatrix {
 *     public int get(int row, int col) {}
 *     public List<Integer> dimensions {}
 * };
 */

class Solution {
    public int leftMostColumnWithOne(BinaryMatrix binaryMatrix) {
        List<Integer> dim = binaryMatrix.dimensions();
        int rows = dim.get(0), cols = dim.get(1);
        int row = 0, col = cols-1;
        int res = -1;
        
        while(row < rows && col >= 0){
            if(binaryMatrix.get(row, col) == 1){
                res = col--;
            }else{
                row++;
            }
        }
        
        return res;
        
    }
}

File: Python3/1428. Leftmost Column with at Least a One.py
# """
# This is BinaryMatrix's API interface.
# You should not implement it, or speculate about its implementation
# """
#class BinaryMatrix(object):
#    def get(self, row: int, col: int) -> int:
#    def dimensions(self) -> list[]:

class Solution:
    def leftMostColumnWithOne(self, binaryMatrix: 'BinaryMatrix') -> int:
        rows, cols = binaryMatrix.dimensions()
        # top-right corner
        row, col = 0, cols-1
        res = -1
        
        while row < rows and col >= 0:
            # found 1, move left
            if binaryMatrix.get(row, col):
                res = col
                col -= 1
            # found 0, move down
            else:
                row += 1
                
        
        return res